core/src/main/java/org/infinispan/reactive/RxJavaInterop.java

```diff
@@ -124,4 +134,35 @@ private RxJavaInterop() { }
    };
 
    private static final Function<Map.Entry<Object, Object>, Object> entryToKeyFunction = Map.Entry::getKey;
+
+   private static final java.util.function.Function<CompletionStage<?>, Completable>
+      completionStageCompletableFunction = CompletionStageCompletableAdapter::new;
```

Comment:
I think using the following might be a bit more clear.

```
  = stage -> {
        CompletableSubject cs = CompletableSubject.create();

        future.whenComplete((v, e) -> {
            if (e != null) {
                cs.onError(e);
            } else {
                cs.onComplete();
            }
        });

        return cs;
     }
```

Reply:
I didn't know about `CompletableSubject`, thanks.

BTW I made it a lambda to match the others, but I don't really need a function object, and I noticed that most callers of the other functions in `RxJavaInterop` also call `apply()` immediately.

Reply:
An example is https://github.com/infinispan/infinispan/blob/0d56cc6f8c1556b71df6c1add0d5bf8ab4b68a38/core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java#L382 So if you don't expect to do that, I wouldn't worry about adding the Function.

Reply:
I can't make it a regular function, I'd have to move it to another class or give it a new name and I can't think of a new one to explain that it's not like the others.

I knew about `to()`, and I know explicitly caching the lambdas means there's only one callsite/anonymous class/classloader, but I still don't like the explicit `apply()` :)